nexus-sdk/src/main/java/io/nexusrpc/OperationHandler.java

```diff
@@ -0,0 +1,228 @@
+package io.nexusrpc;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.function.BiFunction;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * Handler for an operation.
+ *
+ * <p>An instance of this must be returned from an {@link OperationImpl} annotated method in a
+ * {@link ServiceImpl}. Simple synchronous operations can use {@link #sync}, more complex or
+ * asynchronous operations can manually implement this interface.
+ *
+ * @param <T> The parameter type of the operation. This can be {@link Void} for no parameter.
+ * @param <R> the return type of the operation. This can be {@link Void} for no return.
+ */
+public interface OperationHandler<T, R> {
+  /**
+   * Create an operation handler for a synchronous operation backed by the given function. This
+   * function will be called for each operation invocation. All other calls to the operation are not
+   * supported.
+   */
+  static <T, R> OperationHandler<T, R> sync(BiFunction<StartContext, @Nullable T, R> func) {
```

Comment:
We ended up calling this `Operation` in Go and have a different concept of a `Handler` that is not bound to a specific operation or service. I think what you have is fine but we should be consistent in the terminology and modify Go if this is the direction we're heading.

Reply:
I think the langs can be different here. `Operation` defines an operation (annotation) and a `OperationHandler` is the implementation of it (interface). They can't really be the same thing in Java. An `OperationHandler` is completely unrelated to the `Handler` concept in Go (and as the HTTP portion is implemented, we may very well have a `Handler` in Java that matches what Go does).

Reply:
I think specifically here we want consistency.

Reply:
These are basic nexus concepts that should not need a per language exception in the guide.